Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/booster/BoosterRulesRepository.kt

```diff
@@ -41,32 +43,46 @@ class BoosterRulesRepository @Inject constructor(
     val rules: Flow<List<DccValidationRule>> = internalData.data
 
     /**
-     * This updates the booster notification rules and returns true if new rules were fetched from the server.
+     * This updates the booster notification rules.
      * Falls back to previous cached rules in case of an error.
      * Worst case is an empty list.
+     *
+     * @return UpdateResult.UPDATE if new booster rules got downloaded from the server, UpdateResult.NO_UPDATE when
+     * there were no new rules from the server, or UpdateResult.FAIL if the request or parsing failed.
      */
-    suspend fun update(): Boolean {
+    suspend fun update(): UpdateResult {
         Timber.tag(TAG).d("updateBoosterNotificationRules()")
 
-        var updated = false
+        var updateResult = UpdateResult.NO_UPDATE
 
         internalData.updateBlocking {
             return@updateBlocking try {
                 val ruleSetResult = server.ruleSetJson(DccValidationRule.Type.BOOSTER_NOTIFICATION)
-                if (ruleSetResult.source == DccValidationServer.RuleSetSource.SERVER) {
-                    updated = true
+                when (ruleSetResult.source) {
+                    SERVER -> {
+                        updateResult = UpdateResult.UPDATE
+                        ruleSetResult.ruleSetJson.toRuleSet()
+                            .also { localCache.saveBoosterNotificationRulesJson(ruleSetResult.ruleSetJson) }
+                    }
+                    CACHE -> {
+                        updateResult = UpdateResult.NO_UPDATE
+                        ruleSetResult.ruleSetJson.toRuleSet()
+                    }
                 }
-                ruleSetResult.ruleSetJson.toRuleSet()
-                    .also { localCache.saveBoosterNotificationRulesJson(ruleSetResult.ruleSetJson) }
             } catch (e: Exception) {
+                updateResult = UpdateResult.FAIL
                 Timber.tag(TAG).w(e, "Updating booster notification rules failed, loading cached rules")
                 localCache.loadBoosterNotificationRulesJson().toRuleSet()
             }
         }.let { boosterNotificationRules ->
             boosterNotificationRules.also { Timber.tag(TAG).d("Booster notification rules size=%s: %s", it.size, it) }
         }
 
-        return updated
+        return updateResult
+    }
+
+    enum class UpdateResult {
+        UPDATE, NO_UPDATE, FAIL
```

Comment:
can we use one `UpdateResult`?

Reply:
Yes. Created an enum in `utils.repositories`. I hope this is fine for you?